SimpleExec/ProcessExtensions.cs

```diff
@@ -12,13 +13,31 @@ public static void Run(this Process process, bool noEcho)
             process.WaitForExit();
         }
 
-        public static async Task RunAsync(this Process process, bool noEcho)
+        public static async Task RunAsync(this Process process, bool noEcho, CancellationToken cancellationToken = default(CancellationToken))
         {
             process.EnableRaisingEvents = true;
             var tcs = new TaskCompletionSource<object>();
             process.Exited += (s, e) => tcs.SetResult(null);
             process.EchoAndStart(noEcho);
-            await tcs.Task.ConfigureAwait(false);
+            using (KillProcessOnCancellation(process, cancellationToken))
+            {
+                await tcs.Task.ConfigureAwait(false);
+            }
+        }
+
+        static CancellationTokenRegistration KillProcessOnCancellation(Process process, CancellationToken cancellationToken)
+        {
+            return cancellationToken.Register(
+                state =>
+                {
+                    var currentProcess = (Process)state;
+                    if (!currentProcess.HasExited)
+                    {
+                        currentProcess.Kill();

```

Comment:
What happens if the process exits after line 34 has executed, but before this line is executed?

Reply:
Good question - Likely we'll see an exception here. I'll look into redesigning how this works.

Reply:
I've got this resolved with two separate tactics
1. Using the `Process.Exited` event handler to cancel the registration. This will prevent having to attempt `Process.Kill` in a `try/catch`
2. Fail safe to catch exceptions that `Process.Kill` throws during the event that the process is in the process of or is terminating.